test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+  text_payload.resize(5);
+  (*(frames.value()[0].payload_)).copyOut(0, 5, text_payload.data());
+  // Unmasking the text payload
+  for (size_t i = 0; i < text_payload.size(); ++i) {
+    text_payload[i] ^= (frames.value()[0].masking_key_.value() >> (8 * (3 - i % 4))) & 0xff;
+  }
+  EXPECT_EQ("Hello", text_payload);
+}
+
+TEST(WebSocketCodecTest, decodeValidMultipleMaskedFrames) {
+  Encoder encoder;
+  Buffer::OwnedImpl buffer;
+  std::vector<uint8_t> header;
+  std::vector<uint8_t> payload;
+
+  // "hello "
+  header = encoder.encodeFrameHeader({false, 1, 0xaf4be87a, 6, nullptr});
+  payload = {0xc7, 0x2e, 0x84, 0x16, 0xc0, 0x6b};
+  buffer.add(header.data(), header.size());
+  buffer.add(payload.data(), payload.size());
+
+  // " from"
+  header.clear();
+  payload.clear();
+  header = encoder.encodeFrameHeader({false, 0, 0x3c332a16, 5, nullptr});
+  payload = {0x1c, 0x55, 0x58, 0x79, 0x51};
+  buffer.add(header.data(), header.size());
+  buffer.add(payload.data(), payload.size());
+
+  // "client"
+  header.clear();
+  payload.clear();
+  header = encoder.encodeFrameHeader({true, 0, 0x7c96263f, 6, nullptr});
+  payload = {0x1f, 0xfa, 0x4f, 0x5a, 0x12, 0xe2};
+  buffer.add(header.data(), header.size());
+  buffer.add(payload.data(), payload.size());
+
+  Decoder decoder;
+  absl::optional<std::vector<Frame>> frames = decoder.decode(buffer);
+
+  ASSERT_TRUE(frames.has_value());
+  EXPECT_EQ(3, frames.value().size());
+
+  EXPECT_EQ(true, frames.value()[0].masking_key_.has_value());
+  EXPECT_EQ(0xaf4be87a, frames.value()[0].masking_key_.value());
+  EXPECT_EQ(6, frames.value()[0].payload_length_);
+
+  EXPECT_EQ(true, frames.value()[1].masking_key_.has_value());
+  EXPECT_EQ(0x3c332a16, frames.value()[1].masking_key_.value());
+  EXPECT_EQ(5, frames.value()[1].payload_length_);
+
+  EXPECT_EQ(true, frames.value()[2].masking_key_.has_value());
+  EXPECT_EQ(0x7c96263f, frames.value()[2].masking_key_.value());
+  EXPECT_EQ(6, frames.value()[2].payload_length_);
+
+  std::string text_payload;
+
+  text_payload.resize(6);
+  (*(frames.value()[0].payload_)).copyOut(0, 6, text_payload.data());
+  // Unmasking the text payload
+  for (size_t i = 0; i < text_payload.size(); ++i) {
+    text_payload[i] ^= (frames.value()[0].masking_key_.value() >> (8 * (3 - i % 4))) & 0xff;
+  }
+  EXPECT_EQ("hello ", text_payload);
+
+  text_payload.resize(5);
+  (*(frames.value()[1].payload_)).copyOut(0, 5, text_payload.data());
+  // Unmasking the text payload
+  for (size_t i = 0; i < text_payload.size(); ++i) {
+    text_payload[i] ^= (frames.value()[1].masking_key_.value() >> (8 * (3 - i % 4))) & 0xff;
+  }
+  EXPECT_EQ(" from", text_payload);
+
+  text_payload.resize(6);
+  (*(frames.value()[2].payload_)).copyOut(0, 6, text_payload.data());
+  // Unmasking the text payload
+  for (size_t i = 0; i < text_payload.size(); ++i) {
+    text_payload[i] ^= (frames.value()[2].masking_key_.value() >> (8 * (3 - i % 4))) & 0xff;
+  }
+  EXPECT_EQ("client", text_payload);
```

Comment:
Let's remove all of the test code which does unmasking since that's not actually testing any production code.